dts: arm: st: add support for STM32F205xx SOC

Mforcen opened the pull request:
Signed-off-by: Manuel Forcén Muñoz <manuforcen@gmail.com>

STM32F205xx device tree include files added.
Tested properly in a custom board.

gmarull reviewed (changes requested):
Thanks for this addition. Please, make clock fix part of one commit and new SoC addition part of another one. Commit message can also be improved, e.g. `dts: arm: st: add support for xxx`. https://docs.zephyrproject.org/latest/contribute/index.html

Mforcen committed:
dts: arm: st: add support for STM32F205xx SOC

STM32F205xx device tree include files added.
Tested properly in a custom board.

Signed-off-by: Manuel Forcén Muñoz <manuforcen@gmail.com>

Mforcen commented on dts/arm/st/f2/stm32f2.dtsi:
```diff
@@ -172,7 +172,7 @@
 		usart1: serial@40011000 {
 			compatible = "st,stm32-usart", "st,stm32-uart";
 			reg = <0x40011000 0x400>;
-			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00004000>;
```
This was changed in the latedt commit

erwango commented on dts/arm/st/f2/stm32f2.dtsi:
```diff
@@ -172,7 +172,7 @@
 			compatible = "st,stm32-usart", "st,stm32-uart";
 			reg = <0x40011000 0x400>;
-			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00004000>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000010>;
```
Already fixed in https://github.com/zephyrproject-rtos/zephyr/pull/32028 based on the issue you reported.
Commit could be removed.

Mforcen replied:
I'm sorry, I'm kind of new to all these things about pull requests.
I'll try my best.
I'm going to remove the clock commit.
